src/backend/distributed/commands/table.c

```diff
@@ -597,7 +600,7 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 	 * we have a special implementation for ALTER INDEX, and a specific error
 	 * message in case of unsupported sub-command.
 	 */
-	if (leftRelationKind == RELKIND_INDEX)
+	if (leftRelationKind == RELKIND_INDEX || leftRelationKind == RELKIND_PARTITIONED_INDEX)
```

Comment:
There are several other references to `RELKIND_INDEX` in the citus codebase. For example, should we update `PreprocessDropIndexStmt` -> `RangeVarCallbackForDropIndex` to handle partitioned indexes in a more explicit way (e.g., we seem to blindly use `RELKIND INDEX`)? 

Any other places? I think `citus_table_is_visible` should be updated, and we can even add a test.

Reply:
I've added test for ErrorIfUnsupportedAlterIndexStmt and RELKIND_PARTITIONED_INDEX to visibility related parts. Since we have explicit check for RELKIND_PARTITIONED_INDEX on RangeVarCallbackForDropIndex, I think we are ok. I'll add related test(s).

Reply:
Since we couldn't reproduce 'DROP INDEX CONCURRENTLY ... " issue again, we've decided to skip this for this PR.